### ISSUE TITLE
App crashes on startup on some Samsung phones

### ISSUE THREAD

**Issue by robinlinden** [evilcorpltd/aTox#958]
No interesting stack trace, but appears to only happen on Samsung phones:
```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3792)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3968)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:85)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2307)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:246)
  at android.app.ActivityThread.main (ActivityThread.java:8512)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:602)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1130)
```

**Comment by robinlinden** [comment]
Doesn't happen on all Samsung phones. Unable to reproduce on a Samsung Galaxy S8 w/ Android 8.

**Comment by sudden6** [comment]
I'm hit by this :( Samsung Galayy A40, Android 11, January 2022 update.

**Comment by robinlinden** [comment]
That's great! Then I can fix it. :D Think you can get me an [adb log](https://stackoverflow.com/a/9727223)?

**Comment by sudden6** [comment]
[aTox-short.log](https://github.com/evilcorpltd/aTox/files/8053947/aTox-short.log)

Done, I guess this is the key:

```02-11 22:42:48.007 20767 20767 E AndroidRuntime: Caused by: java.lang.SecurityException: ltd.evilcorp.atox was not granted  this permission: android.permission.WRITE_SETTINGS.```

**Comment by robinlinden** [comment]
Thank you! That's incredibly helpful. :D Looks like newer Samsung phones require the WRITE_SETTINGS permission to **read** the current ringtone. I'll have a fix for it in a bit!

The line that crashes:
https://github.com/evilcorpltd/aTox/blob/a05ded2be91f52bfdbd54ceb5123dbfc05ab169a/atox/src/main/kotlin/ui/NotificationHelper.kt#L72

Other people have run into the same fun: `https://github.com/home-assistant/android/issues/1465#issuecomment-821332957`

**Comment by robinlinden** [comment]
sudden6 helped me test the fix in #981 and confirmed it resolved the issue on their device. :P (Thank you again!)
